### PR TITLE
Add `pre-commit` linting with GitHub Actions support

### PR DESCRIPTION
**Pull request by mcdonnnj**
#  #

## 🗣 Description ##

This pull request adds the linting tools (pared down) from our [cisagov/skeleton-generic](https://github.com/cisagov/skeleton-generic) project.

## 💭 Motivation and context ##

This will allow anyone submitting a pull request to test that their submissions meets formatting standards locally. With the GitHub Actions workflow it will also ensure that before a pull request is merged it meets those same standards.

## 🧪 Testing ##

This branch passes the newly added automated testing.

## ✅ Checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - _eschew scope creep!_
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/product-submission-template.md [hunk 1]
```diff
@@ -1,35 +1,27 @@
...
-name: Software Product Submission Template
-about: Schema for product submission for log4j vulnerability.
+# Submission Template #
 
----
-
-# Submission Template
-
-Please provide the following information.  
+Please provide the following information.
```
Comment (jsf9k) [inline .github/ISSUE_TEMPLATE/product-submission-template.md:8]:
```suggestion
Please provide the following information:
```

File: .github/ISSUE_TEMPLATE/product-submission-template.md [hunk 2]
```diff
@@ -1,35 +1,27 @@
...
 
 - Vendor Name
 - Product Name
 - Version(s) affected
-- Status:  Please choose from one of the following (Unknown/Affected/Not Affected/Fixed/Under Investigation).
-- Update Available:  Yes or No (If Yes, please provide link to information)
+- Status: Please choose from one of the following - Unknown, Affected,
+  Not Affected, Fixed, and Under Investigation.
+- Update Available: Yes or No (If Yes, please provide link to information)
```
Comment (jsf9k) [inline .github/ISSUE_TEMPLATE/product-submission-template.md:15]:
```suggestion
- Update Available: Yes or No (If Yes, please provide link to information.)
```

File: .github/ISSUE_TEMPLATE/product-submission-template.md [hunk 3]
```diff
@@ -1,35 +1,27 @@
...
 - Notes
 - References
-- Last Updated:  Date of last update
+- Last Updated: Date of last update
```
Comment (jsf9k) [inline .github/ISSUE_TEMPLATE/product-submission-template.md:18]:
```suggestion
- Last Updated: Date of last update.
```

File: README.md [hunk 4]
```diff
@@ -373,7 +387,7 @@ This list was initially populated using information from the following sources:
 | Fortinet | FortiSwitch & FortiSwitchManager | | Not Affected | | [PSIRT Advisories  FortiGuard](https://www.fortiguard.com/psirt/FG-IR-21-245) | | | |
 | Fortinet | FortiToken Cloud | | Not Affected | | [PSIRT Advisories  FortiGuard](https://www.fortiguard.com/psirt/FG-IR-21-245) | | | |
 | Fortinet | FortiVoice | | Not Affected |  | [PSIRT Advisories  FortiGuard](https://www.fortiguard.com/psirt/FG-IR-21-245) | | | |
-| Fortinet | FortiWeb Cloud | | Not Affected | | [PSIRT Advisories  FortiGuard](https://www.fortiguard.com/psirt/FG-IR-21-245) | | || 
+| Fortinet | FortiWeb Cloud | | Not Affected | | [PSIRT Advisories  FortiGuard](https://www.fortiguard.com/psirt/FG-IR-21-245) | | ||
```
Comment (dav3r) [inline README.md:390]: Clean up whitespace:
```suggestion
| Fortinet | FortiWeb Cloud | | Not Affected | | [PSIRT Advisories FortiGuard](https://www.fortiguard.com/psirt/FG-IR-21-245) | | | |
```

File: README.md [hunk 5]
```diff
@@ -488,7 +502,7 @@ This list was initially populated using information from the following sources:
...
-| Sophos | Cloud Optix | | Fixed | | [Advisory: Log4J zero-day vulnerability AKA Log4Shell (CVE-2021-44228)  Sophos](https://www.sophos.com/en-us/security-advisories/sophos-sa-20211210-log4j-rce) | Users may have noticed a brief outage around 12:30 GMT as updates were deployed.<br>There was no evidence that the vulnerability was exploited and to our knowledge no customers are impacted. | | 12/12/2021   |
+| Sophos | Cloud Optix | | Fixed | | [Advisory: Log4J zero-day vulnerability AKA Log4Shell (CVE-2021-44228)  Sophos](https://www.sophos.com/en-us/security-advisories/sophos-sa-20211210-log4j-rce) | Users may have noticed a brief outage around 12:30 GMT as updates were deployed. There was no evidence that the vulnerability was exploited and to our knowledge no customers are impacted. | | 12/12/2021   |
```
Comment (dav3r) [inline README.md:505]: Remove extra whitespace
```suggestion
| Sophos | Cloud Optix | | Fixed | | [Advisory: Log4J zero-day vulnerability AKA Log4Shell (CVE-2021-44228) Sophos](https://www.sophos.com/en-us/security-advisories/sophos-sa-20211210-log4j-rce) | Users may have noticed a brief outage around 12:30 GMT as updates were deployed. There was no evidence that the vulnerability was exploited and to our knowledge no customers are impacted. | | 12/12/2021 |
```

**Review by jsf9k — APPROVED** [review]
I noticed a few very minor things, but otherwise this looks great to me.

**Review by felddy — APPROVED** [review]
💪 Strong work.

**Review by dav3r — APPROVED** [review]
Looking good - only trivial whitespace issues noted. 👍

**Review by justmurphy — APPROVED** [review]

**Pull request merged**